packages/jaeger-ui/src/utils/readJsonFile.tsx

```diff
@@ -24,7 +24,21 @@ export default function readJsonFile(fileList: { file: File }): Promise<string>
       }
       try {
         const traceObj = JSON.parse(reader.result);
-        if ('resourceSpans' in traceObj) {
+        if (Array.isArray(traceObj) && traceObj.every(obj => 'resourceSpans' in obj)) {
+          const mergedResourceSpans = traceObj.reduce((acc, obj) => {
+            acc.push(...obj.resourceSpans);
+            return acc;
+          }, []);
+
+          const mergedTraceObj = { resourceSpans: mergedResourceSpans };
+          JaegerAPI.transformOTLP(mergedTraceObj)
```

Comment:
Instead you can reassign the result to trace obj and end the if, so that it's further processed by the next if.

Reply:
Sure, I'll do it Asap!